leshan-server-cf/src/main/java/org/eclipse/leshan/server/californium/LeshanServerBuilder.java

```diff
@@ -392,14 +411,28 @@ public LeshanServer build() {
         }
 
         // create endpoints
-        CoapEndpoint unsecuredEndpoint;
-        CoapEndpoint securedEndpoint;
-        if (endpointFactory != null) {
-            unsecuredEndpoint = endpointFactory.createUnsecuredEndpoint(localAddress, coapConfig, registrationStore);
-            securedEndpoint = endpointFactory.createSecuredEndpoint(dtlsConfig, coapConfig, registrationStore);
-        } else {
-            unsecuredEndpoint = new CoapEndpoint(localAddress, coapConfig, registrationStore);
-            securedEndpoint = new CoapEndpoint(new DTLSConnector(dtlsConfig), coapConfig, registrationStore, null);
+        CoapEndpoint unsecuredEndpoint = null;
+        if (!noUnsecuredEndpoint) {
+            if (endpointFactory != null) {
+                unsecuredEndpoint = endpointFactory.createUnsecuredEndpoint(localAddress, coapConfig,
+                        registrationStore);
+            } else {
+                unsecuredEndpoint = new CoapEndpoint(localAddress, coapConfig, registrationStore);
+            }
+        }
+
+        CoapEndpoint securedEndpoint = null;
+        if (!noSecuredEndpoint && securityStore != null) {
```

Comment:
We should document somewhere that the securityStore is mandatory to start the secured endpoint.